packages/x-data-grid/src/hooks/features/editing/useGridCellEditing.ts

```diff
@@ -127,8 +127,7 @@ export const useGridCellEditing = (
     (params, event) => {
       if (params.cellMode === GridCellModes.Edit) {
         // Wait until IME is settled for Asian languages like Japanese and Chinese
-        // TODO: `event.which` is deprecated but this is a temporary workaround
-        if (event.which === 229) {
+        if (event.type === 'compositionstart' || event.type === 'compositionupdate') {
```

Comment:
I don't see how this can work. I'm not aware that those types exist: https://w3c.github.io/uievents/#events-keyboard-types.

Reply:
Potentially, in the future we could rely on this: https://github.com/mui/material-ui/pull/39713#discussion_r1381678957.

Reply:
> I don't see how this can work. I'm not aware that those types exist: https://w3c.github.io/uievents/#events-keyboard-types.

they exist, you can find them here : https://w3c.github.io/uievents/#events-composition-types

Reply:
> Potentially, in the future we could rely on this: [mui/material-ui#39713 (comment)](https://github.com/mui/material-ui/pull/39713#discussion_r1381678957).

I have decided to use `compositionstart` and `compositionupdate` event types because they are supported in older browsers as well.
but yes we can use `isComposing` if we have dropped the support for older browsers (that's a future thing as you also mentioned )

Reply:
@k-rajat19 Ok, so this logic does nothing. For `event.type === 'compositionstart'` to be ever true, we would need `'cellKeyDown'` to listen to the `compositionstart` event which it doesn't.